### ISSUE TITLE
Remove AppCenter telemetry

### ISSUE THREAD

**Issue by dotMorten** [Azure/azure-spatial-anchors-samples#31]
I noticed that the Android Java SDK takes a dependency on AppCenter and is reporting telemetry.
Please remove this, as SDK's should never collect telemetry without app developers being able to opt in to (and not out of) to it. They could inadvertently be breaking data privacy policies, because we have no knowledge this is happening under our noses.

IMHO it's a big no-no for SDKs to collect telemetry. That's for app-developers to do only.

**Comment by dotMorten** [comment]
The latest xamarin package also forces you to pull in analytics. This is a showstopper for me, as I have no control over what private IP is being sent to who and where, and I wouldn't know what to put in my privacy policy, or know how to disable it all. Again telemetry only belong in apps, not SDKs.

![image](https://user-images.githubusercontent.com/1378165/70942229-0dfe9400-2003-11ea-93e6-204824a648c5.png)

**Comment by stef-t** [comment]
Is there an update on the _Spatial Anchors SDK_ sending telemetry? Unfortunately I cannot find any information regarding this topic in, e.g, the FAQs and I would like to know what is send and if I can opt-in or opt-out.

On Android an _appcenter-analytics_ dependency is defined in the gradle:
> implementation('com.microsoft.appcenter:appcenter-analytics:[1.10.0]')

On iOS, I could not find a similar explicit dependency. 

As @dotMorten mentioned, it's important for an app's privacy policy that uses the _Spatial Anchors SDK_. Now more so as, e.g., Apple requires [_Privacy Labels_](https://developer.apple.com/app-store/app-privacy-details/) from businesses for each app the publish. Failing to disclose the correct information is reason for removal of an app from the AppStore.

**Comment by msftradford** [comment]
Thanks for reaching out @dotMorten and @stef-t! We have opened an internal investigation (#31806367) to get answers to these questions. We'll provide updates here when we have them. Thanks!

**Comment by anod** [comment]
@msftradford Any update on the issue, the issue is blocking from integration of the SDK in Android:

1. The library used is outdated (1.10.0) and conflicts with newer version used in our app (4.2.0)
2. Because there is no way to disable it, this breaks privacy settings in our app

**Comment by mrwellmann** [comment]
> 1. The library used is outdated (1.10.0) and conflicts with newer version used in our app (4.2.0)

@anod we had the same and could fix it with a modification to the mainTemplate.gradle. You can find the solution here https://github.com/Azure/azure-spatial-anchors-samples/issues/272#issuecomment-882657279

**Comment by MehranAzimi-msft** [comment]
@anod and @stef-t As you have discussed, ASA SDK collects telemetry using AppCenter SDK, and we are considering moving away to other solutions that should solve the build issue/conflicts you have reported here. 

Besides the conflict issue with AppCenter, can you please elaborate more on this:
> Now more so as, e.g., Apple requires Privacy Labels from businesses for each app the publish. Failing to disclose the correct information is reason for removal of an app from the AppStore.

Are you only asking for a statement about what telemetry is collected by ASA SDK so you can include it in your app privacy labels? Is your app required to have the option to shut off all telemetry, and if yes, where is this requirement coming from?

Thanks

**Comment by dotMorten** [comment]
> Are you only asking for a statement about what telemetry is collected by ASA SDK so you can include it in your app privacy labels

No we are asking you to stop collecting telemetry. Telemetry should not be in an SDK reference. You're setting us up to fail.

**Comment by stef-t** [comment]
For my part, in a first step, I'm asking for more information on what telemtry is actually collected, such that this information can be included in a privacy statement.

_In a second step, an opt out switch is mandatory._

I understand that telemetry is important and is part of how things are done, but I think you have to look at this from an app publisher's perspective. 
Data is a _very sensitive_ topic these days! Transparency in regards to data collection is the way to go to show your customers you take them seriously. This is what privacy statements are for. On the Apple side of things the privacy labels offer a simplified visualization of the privacy statement that makes sense to most users. 
Handling and collection of data, when an app is used, more and more influences a customers decision to choose and install an app.

Publishers of apps that want to use the ASA SDK, have a very hard time reasoning about their app's data collection, when the ASA SDK sends _currently still unknown (!)_ telemetry and there is no option to turn it off. This way the ASA SDK imposes non-technical restrictions on an app, while there exist a "simple" solution to this: Add an opt out switch.

**Comment by MehranAzimi-msft** [comment]
Thanks for your responses. We are considering addressing these issues in next ASA SDK release per your feedback, and will keep this thread updated.

**Comment by phamqduc** [comment]
We would like to update that ASA SDK 2.11.0 was released recently that adds a new CloudSpatialAnchorSession::EnableTelemetry method to the SDK API set which allows apps to **disable** ASA SDK Telemetry. Please see more details are in the [release notes](https://github.com/Azure/azure-spatial-anchors-samples/releases/tag/v2.11.0). Please let us know if this may help to address the issues/concerns here.

**Comment by anod** [comment]
Thanks @phamqduc for the update, just checked andorid skd and it still depends on AppCenter

```
+--- com.microsoft.azure.spatialanchors:spatialanchors_jni:2.11.0
|    +--- com.microsoft.azure.spatialanchors:spatialanchors_java:2.11.0
|    |    +--- com.squareup.okhttp3:okhttp:3.11.0 -> 4.8.0 (*)
|    |    +--- com.google.ar:core:1.11.0 -> 1.25.0
|    |    \--- com.microsoft.appcenter:appcenter-analytics:1.10.0
|    |         \--- com.microsoft.appcenter:appcenter:1.10.0 -> 4.2.0
|    +--- com.squareup.okhttp3:okhttp:3.11.0 -> 4.8.0 (*)
````

**Comment by shreya-sharma** [comment]
Hello, thanks for pointing this out. We see that this is a bug and we have filed an internal bug (#37215814) to track it and make sure that call site is cleaned up too for when setTelemetryEnabled is set to False. 
As for the Android still dependent on the App Center, you are correct, we have only moved our iOS bits to 1DS in the last release. Rest assured we are working on moving Android away from AppCenter too and should have that fixed in a future release.

**Comment by wyliefoxxx** [comment]
I've been experimenting with other issues, and playing with the effects of disabling Telemetry. I'd love some clarification as to what the effects of enabling/disabling telemetry are. I can see no performance difference, but I'm afraid having it turned off might have an impact long-term performance for a number of hypothetical reasons... Is there any way we (everyone) can get some more information on what the costs and benefits of sending telemetry data are? Either here, or somewhere in the documentation?

**Comment by dotMorten** [comment]
@wyliefoxxx why would disabling it cause a performance degradation? It is doing less (as in not shadily tracking you and reporting to Microsoft what you’re doing)

**Comment by Varnauld** [comment]
Hello - we released v2.13.0 mid-July https://github.com/Azure/azure-spatial-anchors-samples/releases - this includes AppCenter dependency removal from all platforms. Please let us know if anything further is required or if you have any feedback you would like to share. Thank you,
